ansible_ai_connect/users/views.py

```diff
@@ -59,6 +59,26 @@ def __init__(self, *args, **kwargs):
         except WcaSecretManagerMissingCredentialsError:
             pass
 
+        # Check authentication has bene configured
+        self.is_auth_configured = True
+        match settings.DEPLOYMENT_MODE:
+            case "saas":
+                self.is_auth_configured = (
+                    settings.SOCIAL_AUTH_OIDC_OIDC_ENDPOINT
+                    and settings.SOCIAL_AUTH_OIDC_KEY
+                    and settings.SOCIAL_AUTH_OIDC_SECRET
+                )
+            case "onprem":
+                self.is_auth_configured = (
+                    settings.AAP_API_URL
+                    and settings.SOCIAL_AUTH_AAP_KEY
+                    and settings.SOCIAL_AUTH_AAP_SECRET
+                )
+            case "upstream":
+                self.is_auth_configured = (
+                    settings.SOCIAL_AUTH_GITHUB_KEY or settings.SOCIAL_AUTH_GITHUB_TEAM_KEY
+                )
```

Comment:
I believe it would be more elegant to have a `case _:` and set `self.is_auth_configured = True` there.

Reply:
Also, it's a situation that should never happen. Instead of trying to deal with it, we should IMO instead raise a `ValueError` exception instead.

Reply:
@goneri Let me try to understand.

Is `DEPLOYMENT_MODE == "upstream"` obsolete?

Are the various `SOCIAL_AUTH_GITHUB_xxx` settings obsolete?

I'd rather create a _maintenance_ JIRA to remove it entirely than raise an exception.